### DISCUSSION TITLE
Custom Build Commands

### DISCUSSION THREAD

**Discussion by jparr721** [microsoft/vscode-cpptools#7138] [Q&A]
Inside of my current CMake configuration, I can only get it working with a particular command sequence:
```
cmake -GNinja -DCMAKE_APPLE_SILICON_PROCESSOR=arm64
```
When I put this configuration into vscode, it generates the following command on startup:

```
/opt/homebrew/bin/cmake --no-warn-unused-cli -DCMAKE_APPLE_SILICON_PROCESSOR:STRING=arm64 -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -DCMAKE_C_COMPILER:FILEPATH=/usr/bin/clang -DCMAKE_CXX_COMPILER:FILEPATH=/usr/bin/clang++ -H/Users/jarred/Code/qt_lattice -B/Users/jarred/Code/qt_lattice/build -G Ninja
```

Is there a way to override this functionality? For some reason, when it generates this command it causes my project to begin throwing `undefined symbol for architecture` errors. Personally, I'd rather just debug it on my own if I break something without all the extra commands. The documentation was not very clear on how I can just pass in a string to be run instead of the generated string here.

Oddly enough, when building for Xcode, everything works perfectly and compiles without the aforementioned issues. Are the extra flags necessary? How can I prune them all from the command string to allow myself to be able to just run _exactly_ what I want it to run?

Thanks in advance!

**Comment by sean-mcmanus** [comment]
Is this related to the C/C++ extension or the CMake Tools extension or CMake itself? I don't know what "put this configuration into vscode" means.

**Reply by jparr721** [reply]
The CMake Tools extension. I appear to have clicked the wrong link and erroneously posted here, however. Sorry about that!

**Reply by bobbrow** [reply]
There was an issue opened in CMake Tools that sounds related.  https://github.com/microsoft/vscode-cmake-tools/issues/1639
You may also be able to get the COMPILER variables removed by selecting the "[Unspecified]" Kit.
